### PR TITLE
Add Mirror link for Dilar "ar"

### LINKED ISSUE #1945: Dilar : mirror url
### Source information

Banned from some countries

### Source language

Dilar 1.4.2

### Source new URL

https://golden.rest/

### Other details

This is a second link to the site to work in other countries

### Acknowledgements

- [X] I have updated all installed extensions.
- [X] I have opened WebView and checked that the source URL has not changed yet.
- [X] I have searched the existing issues and this is a new ticket, **NOT** a duplicate or related to another open or closed issue.
- [X] I have written a short but informative title.
- [ ] I have admitted that I am a clown by having checked this box, as I have not read these acknowledgements.
- [X] I will fill out all of the requested information in this form.

### PR DESCRIPTION
**Pull request by mohamedotaku**
Close #1945
Checklist:

- [ ] Updated `extVersionCode` value in `build.gradle` for individual extensions
- [X] Updated `overrideVersionCode` or `baseVersionCode` as needed for all multisrc extensions
- [X] Referenced all related issues in the PR body (e.g. "Closes #xyz")
- [ ] Added the `isNsfw = true` flag in `build.gradle` when appropriate
- [X] Have not changed source names
- [ ] Have explicitly kept the `id` if a source's name or language were changed
- [X] Have tested the modifications by compiling and running the extension through Android Studio
- [ ] Have removed `web_hi_res_512.png` when adding a new extension

### CODE REVIEW

**Review by AwkwardPeak7 — CHANGES_REQUESTED** [review]
you would need to override the `cdnUrl` too.

remove baseUrl and cdnUrl from gmanga constructor and put cdnUrl as `protected open val`, then you can simply override it in dilar.

File: lib-multisrc/gmanga/build.gradle.kts [hunk 1]
```diff
@@ -2,4 +2,4 @@ plugins {
     id("lib-multisrc")
 }
 
-baseVersionCode = 2
+baseVersionCode = 3
```
Comment (AwkwardPeak7) [inline lib-multisrc/gmanga/build.gradle.kts]: no real functionality changed in gmanga, so don't bump

File: src/ar/dilar/src/eu/kanade/tachiyomi/extension/ar/dilar/Dilar.kt [hunk 2]
```diff
@@ -23,4 +38,34 @@ class Dilar : Gmanga(
...
+    override val baseUrl by lazy {
+        when {
+            System.getenv("CI") == "true" -> MIRROR_PREF_ENTRY_VALUES.joinToString("#, ")
+            else -> preferences.getString(MIRROR_PREF_KEY, MIRROR_PREF_DEFAULT_VALUE)!!
+        }
+    }
+
+    private val preferences: SharedPreferences by lazy {
+        Injekt.get<Application>().getSharedPreferences("source_$id", 0x0000)
+    }
```
Comment (AwkwardPeak7) [inline src/ar/dilar/src/eu/kanade/tachiyomi/extension/ar/dilar/Dilar.kt]: remove lazy since you are using it immediately (in cdnUrl, or make that lazy)

**Review by AwkwardPeak7 — CHANGES_REQUESTED** [review]

File: src/ar/dilar/src/eu/kanade/tachiyomi/extension/ar/dilar/Dilar.kt [hunk 3]
```diff
@@ -23,4 +38,30 @@ class Dilar : Gmanga(
...
+            setOnPreferenceChangeListener { _, _ ->
+                Toast.makeText(screen.context, RESTART_TACHIYOMI, Toast.LENGTH_LONG).show()
+                true
+            }
+        }
+        screen.addPreference(mirrorPref)
+    }
+
+    private fun MirrorPref(): String = preferences.getString(MIRROR_PREF_KEY, MIRROR_PREF_DEFAULT_VALUE)!!
+    override val baseUrl = MirrorPref()
```
Comment (AwkwardPeak7) [inline src/ar/dilar/src/eu/kanade/tachiyomi/extension/ar/dilar/Dilar.kt]: you removed the CI thing...

File: src/ar/dilar/src/eu/kanade/tachiyomi/extension/ar/dilar/Dilar.kt [hunk 4]
```diff
@@ -23,4 +38,34 @@ class Dilar : Gmanga(
...
+    private fun mirrorPref() = when {
+        System.getenv("CI") == "true" -> MIRROR_PREF_ENTRY_VALUES.joinToString("#, ")
+        else -> preferences.getString(MIRROR_PREF_KEY, MIRROR_PREF_DEFAULT_VALUE)!!
+    }
+
+    override val baseUrl = mirrorPref()
+
+    private val preferences: SharedPreferences by lazy {
+        Injekt.get<Application>().getSharedPreferences("source_$id", 0x0000)
+    }
```
Comment (AwkwardPeak7) [inline src/ar/dilar/src/eu/kanade/tachiyomi/extension/ar/dilar/Dilar.kt]:
```suggestion
    private val preferences: SharedPreferences =
        Injekt.get<Application>().getSharedPreferences("source_$id", 0x0000)
```

File: src/ar/dilar/src/eu/kanade/tachiyomi/extension/ar/dilar/Dilar.kt (hunk 3 above)
Comment (cuong-tran) [reply src/ar/dilar/src/eu/kanade/tachiyomi/extension/ar/dilar/Dilar.kt]: @AwkwardPeak7 could you please explain what is the purpose of that CI thing?

**Pull request merged**